projects/plugins/jetpack/extensions/blocks/premium-content/login-button/login-button.php

```diff
@@ -31,6 +32,56 @@ function register_login_button_block() {
 }
 add_action( 'init', __NAMESPACE__ . '\register_login_button_block' );
 
+/**
+ * Returns current URL.
+ *
+ * @return string
+ */
+function get_current_url() {
+	if ( ! isset( $_SERVER['HTTP_HOST'] ) || ! isset( $_SERVER['REQUEST_URI'] ) ) {
+		return '';
+	}
+
+	return ( is_ssl() ? 'https://' : 'http://' ) . wp_unslash( $_SERVER['HTTP_HOST'] ) . wp_unslash( $_SERVER['REQUEST_URI'] ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+}
+
+/**
+ * Returns subscriber log in URL.
+ *
+ * @param string $redirect Path to redirect to on login.
+ *
+ * @return string
+ */
+function get_subscriber_login_url( $redirect ) {
```

Comment:
Next up this would be good to move to a central place and share between the login block and this block:

https://github.com/Automattic/jetpack/blob/9bf534ca0f1524ada48b6fdd083135649ba17f4e/projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php#L77-L103